sklearn/naive_bayes.py

```diff
@@ -552,7 +552,7 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
         if _check_partial_fit_first_call(self, classes):
             # This is the first call to partial_fit:
             # initialize various cumulative counters
-            n_effective_classes = len(classes) if len(classes) > 1 else 2
```

Comment:
There is a small behavior change here when `len(classes) == 1`.
However, the docstring implies that `len(classes)` should be 2 for binary problems:
```
        classes : array-like of shape (n_classes,), default=None
            List of all the classes that can possibly appear in the y vector.
```
Assuming this is true, then I believe the only behavior change is for the degenerate single-class case, as intended.